Presentation.Web/app/components/data-processing/tabs/data-processing-registration-tab-main.controller.ts

```diff
@@ -79,39 +81,51 @@
 
-        private bindDataResponsible() {
-            const optionMap = this.dataProcessingRegistrationOptions.dataResponsibleOptions.reduce((acc, next, _) => {
-                acc[next.id] = {
-                    text: next.name,
-                    id: next.id,
-                    optionalObjectContext: {
-                        id: next.id,
-                        name: next.name,
-                        description: next.description
-                    }
-                };
-                return acc;
-            }, {});
-
-            //If selected state is expired, add it for presentation reasons
-            const existingChoice = this.dataProcessingRegistration.dataResponsible.value;
-            if (existingChoice && !optionMap[existingChoice.id]) {
-                optionMap[existingChoice.id] = {
-                    text: `${existingChoice.name} (udgået)`,
-                    id: existingChoice.id,
-                    disabled: true,
-                    optionalObjectContext: existingChoice
-                }
-            }
+        createSubDataProcessor() {
+            this.openModal();
+        }
 
-            const options = this.dataProcessingRegistrationOptions.dataResponsibleOptions.map(option => optionMap[option.id]);
+        updateSubDataProcessor(id: number) {
+            this.openModal(id);
+        }
 
-            this.dataResponsible = {
-                selectedElement: existingChoice && optionMap[existingChoice.id],
-                select2Config: this.select2LoadingService.select2LocalDataNoSearch(() => options, true),
-                elementSelected: (newElement) => this.updateDataResponsible(newElement)
-            };
+        changeName(name) {
+            this.apiUseCaseFactory
+                .createUpdate("Navn", () => this.dataProcessingRegistrationService.rename(this.dataProcessingRegistrationId, name))
+                .executeAsync(nameChangeResponse => {
+                    this.headerName = name;
+                    return nameChangeResponse;
+                });
+        }
+
+        removeSubDataProcessor(id: number) {
+            this.apiUseCaseFactory
+                .createAssignmentRemoval(() => this.dataProcessingRegistrationService.removeSubDataProcessor(this.dataProcessingRegistrationId, id))
+                .executeAsync(success => {
+
+                    //Update the source collection
+                    this.dataProcessingRegistration.subDataProcessors = this.dataProcessingRegistration.subDataProcessors.filter(x => x.id !== id);
+
+                    //Propagate changes to UI binding
+                    this.bindSubDataProcessors();
+                    return success;
+                });
+        }
+
+        private openModal(subDataProcessorId: number = null) {
+            this.$state.go("data-processing.edit-registration.main.sub-data-processor", { subDataProcessorId: subDataProcessorId });
```

Comment:
I am not sure we should change state for this... rather do an in-page modal, like on "FK Org import dialog", where we don't "reload" the state when the dialog closes. Right now the screen flickers and the page is reloaded when using the dialog.. also the vertical position changes